### PR TITLE
enable full text search

### LINKED ISSUE #22: Enable full text search for articles
In order for app users to search for articles properly, it is nessesary, that they can type a text into the search bar, that can not be found in the title of an article, but in the contents of an article.

### PR DESCRIPTION
**Pull request by alexfritsch10**
FIxes #22 (with a test case 🥇)

### CODE REVIEW

File: wiki_service/mongoAPI.py [hunk 1]
```diff
@@ -23,6 +26,7 @@ def __init__(self, ):
         if collection not in db.list_collection_names():
             db.create_collection(collection)
         self.collection = db[collection]
+        #self.collection.create_index([("content", "text")])
```
Comment (isaschm) [inline wiki_service/mongoAPI.py:29]: Can this go?

**Review by isaschm — COMMENTED** [review]
Yayy tests 🎉 
I'm wondering about edge cases though. We should return something when there are no matches.

**Comment by isaschm** [comment]
I had another thought: Adding an extra /search route is not crud anymore. I'm not sure how passionate we are when it comes to sticking to it but maybe we could implement full-text search as a get all with parameters?

**Comment by alexfritsch10** [comment]
If there are no matches for the current search term (e.g. /wiki/search/?query=Mindtastic+Mind) the articles list in the response JSOn is just empty. I think, that this is something that the frontend teams can work with. 

I adjusted the route from /search to /wiki/search. This route hast a path variable named "query".

**Comment by isaschm** [comment]
> If there are no matches for the current search term (e.g. /wiki/search/?query=Mindtastic+Mind) the articles list in the response JSOn is just empty. I think, that this is something that the frontend teams can work with.

Sounds reasonable!

> I adjusted the route from /search to /wiki/search. This route hast a path variable named "query".

I meant like this:
```
@wiki.get('/wiki')
async def wiki_readAllArticles(query: str = None):
    db = MongoAPI()
    if query:
      response = db.searchContent(query)
    else:
      response = db.readAll()
    response["status_code"] = HTTPStatus.OK
    return response
```
